tests/cases/conformance/node/allowJs/nodeModulesAllowJsExportlessJsModuleDetectionAuto.ts

```diff
@@ -0,0 +1,8 @@
+// @module: node16,nodenext
+// @allowJs: true
+// @outDir: ./out
+// @moduleDetection: auto
+// @filename: foo.cjs

```

Comment:
Does this also need change to `isExternalOrCommonJsModule` i think during incremental buildInfo doesnt mark bar.js as affecting global scope and that is a bug ?

Reply:
Afaik that just reads the two marker fields, which are all that get set to determine module-ness.

Reply:
can you add test and ensure that bar.js will be marked as affects global scope and foo.cjs as module so no affectsGlobalScope in buildInfo. I am pretty certain that it is broken right now.